ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -61,6 +64,9 @@ public static KsqlContext create(Map<String, Object> streamsProperties)
     AdminClient adminClient = AdminClient.create(ksqlConfig.getKsqlAdminClientConfigProps());
     KafkaTopicClient topicClient = new KafkaTopicClientImpl(adminClient);
     KsqlEngine ksqlEngine = new KsqlEngine(ksqlConfig, topicClient);
+    Properties supportProperties = new Properties();
+    supportProperties.putAll(streamsProperties);
```

Comment:
Why do you need StreamProperties here?

Reply:
My understanding is anyone using KsqlContext would pass anything required for Ksql through the streams properties. If so, then all all support metrics related configs also go in there. (like turning off metrics, using https, etc.) Off course, this needs to be documented :)